source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -2545,10 +2588,12 @@ export abstract class MediaPlayer {
 export class HTML5MediaPlayer extends MediaPlayer {
     private _selectedMediaType?: string;
     private _selectedSources: MediaSource[] = [];
+    private _captionSources: CaptionSource[] = [];
```

Comment:
From high level, there are two implementations, and implementation 1 is what you did.
Personally, I think 2 is easier to understand and will have less change.
Option 1 is harder to maintain the mapping between source and caption

Implementation 1:
```
"sources": [
				{
					"mimeType": "video/mp4",
					"url": "https://adaptivecardsblob.blob.core.windows.net/assets/AdaptiveCardsOverviewVideo.mp4"
				}
			],
			"captionSources": [
				{
					"mimeType": "vtt",
					"url": "https://gist.githubusercontent.com/PankajBhojwani/d11855a1e0c0b1f70a0e64f81a1fb696/raw/46fa93da5c3703d03843d59cedd8b5f10aa3747b/test.vtt"
				},
				{
					"mimeType": "srt",
					"url": "https://gist.githubusercontent.com/PankajBhojwani/fab75d6493dddd9f5ee964228ca35c71/raw/1706044f46c56216f6a66b8b9e3bb9ecdbf2c56a/zzz.srt"
				}
			]
		}
```

Implementation 2:
```
"sources": [
				{
					"mimeType": "video/mp4",
					"url": "https://adaptivecardsblob.blob.core.windows.net/assets/AdaptiveCardsOverviewVideo.mp4"
					"captionMimeType": "vtt",
					"captionUrl": "https://gist.githubusercontent.com/PankajBhojwani/d11855a1e0c0b1f70a0e64f81a1fb696/raw/46fa93da5c3703d03843d59cedd8b5f10aa3747b/test.vtt"
				}
			]
		}
```

Reply:
If we end up going with this implementation, `captionUrl` will probably need to be `captionSources` and take in an array, because there could be multiple caption tracks for the same video

Reply:
I don't think we can do that. The tracks apply to all the media sources. #2 would imply that each source has its own tracks, which isn't the case.

Reply:
Also, the implementation I followed here is what was outlined in #4963

Reply:
Look like 1 is the way to go. Possible rename CaptionSources to VideoTrack? Also it should include more properties like srclang, kind...

I found an example of multiple sources and tracks from [Adding_captions_and_subtitles_to_HTML5_video](
https://developer.mozilla.org/en-US/docs/Web/Guide/Audio_and_video_delivery/Adding_captions_and_subtitles_to_HTML5_video)

```
<video id="video" controls preload="metadata">
   <source src="video/sintel-short.mp4" type="video/mp4">
   <source src="video/sintel-short.webm" type="video/webm">
   <track label="English" kind="subtitles" srclang="en" src="captions/vtt/sintel-en.vtt" default>
   <track label="Deutsch" kind="subtitles" srclang="de" src="captions/vtt/sintel-de.vtt">
   <track label="Español" kind="subtitles" srclang="es" src="captions/vtt/sintel-es.vtt">
</video>

```

Reply:
@dclaux @licanhua @PankajBhojwani on top of that, we should be able to figure out what **language** each `captionSource` represents. We can enforce the naming of the file url to include the language like `subtitles-en.vtt` for example or add another entry to the schema to indicate the language.

Reply:
@beervoley the spec being referred to up to this PR doesn't mention anything like that. That spec dates back from 2020. Someone should probably revise it and get that revision reviewed before the feature is implemented?

Reply:
@dclaux I agree :) @paulcam206 may be it's a good idea for Anna to look at if it's only adding the language?:)

Reply:
I added a language property, and `label` gets set to that now. `kind` is set to `captions` and according to the [`track` docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/track), `srclang` only needs to be set if `kind` is `subtitles`, but since we are implementing `captions` here, not `subtitles`, we don't need to provide `srclang`. Resolving this thread